theano/tensor/basic.py

```diff
@@ -1427,21 +1427,31 @@ class Argmax(Op):
     nin = 2  # tensor, axis
     nout = 1
     E_axis = 'invalid axis'
-    __props__ = ()
+    __props__ = ('axis',)
```

Comment:
The axis can be
 - NoneConst
 - A theano 1d tensor
so get_scalar_constant_value() will not help.

Most of the code of the op does accesses `node.inputs[1].data`, fully expecting it to be a constant, but never checking therefore it is a constant and trying to treat it like an input is dumb.